jboss/container/wildfly/launch/jgroups/added/launch/ha.sh

```diff
@@ -289,7 +289,7 @@ generate_dns_ping_config() {
 
 configure_ha_args() {
   # Set HA args
-  IP_ADDR=`hostname -i`
+  IP_ADDR=`hostname -i| cut -d " " -f1`
```

Comment:
@buuhsmead , I ma not sure that we can take the first IP address. The server is by default configured to prefer IPv4. In case we have multiple IP addresses we should get the ipv4 I guess. @luck3y could add more there.

We have multiple places where hostname command is used to retrieve the address, seems that we should handle them all.
For example:
https://github.com/wildfly/wildfly-cekit-modules/blob/master/jboss/container/wildfly/launch/messaging/added/launch/messaging.sh#L80
https://github.com/wildfly/wildfly-cekit-modules/blob/master/jboss/container/wildfly/launch/keycloak/added/keycloak.sh#L794 (this one is a left over and should be removed).

Even in the wildfly-s2i project: https://github.com/wildfly/wildfly-s2i/blob/master/wildfly-cloud-legacy-galleon-pack/feature-pack/src/main/resources/packages/wildfly.s2i.common/content/bin/openshift-launch.sh#L8
although this one is overridden by HA args.

Reply:
@buuhsmead +1 to what @jfdenise has said. Your most recent idea will return one address, it may not be the ipv4 address though, which inside of OpenShift is likely the one you want. hostname -I (capital i) will get you some of the way but the ordering is not guaranteed. I'll have to have a think about how we should do this in a safer way.

Reply:
I think that we could detect the ':' character. A IPv4 address will never contain the ':' character.
What do you think of this function that echo the first non ipv6 address it finds:
```
function get_host_name() {
 local host=
 for x in $(hostname -i)
 do
  if [[ $x =~ ":" ]]; then
   continue
  else
   host=$x
   break
  fi
 done
 echo "$host"
}
```
We could add this function to https://github.com/wildfly/wildfly-cekit-modules/blob/master/jboss/container/wildfly/launch-config/os/added/launch/launch-common.sh
And have all places where we call hostname be replaced.

Reply:
@jfdenise I think that looks better, but I think it shouldn't be called get_host_name() as its really looking to return a v4 ip. Perhaps call it get_host_ipv4()? Also we could consider having an env variable for when this doesn't work in some environment and allows the ip / hostname to be set via a var, something like WILDFLY_BIND_HOST_ADDR maybe?